app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordContents.java

```diff
@@ -0,0 +1,354 @@
+                Toast.makeText(context, context.getText(R.string.user_dict_word_modified), Toast.LENGTH_SHORT).show();
+            } else {
+                final String localeInToast = new LocaleRenderer(context, mLocale).toString();
+                final String messageWordAdded = context.getString(R.string.user_dict_word_added) + " " + localeInToast;
+                Toast.makeText(context, messageWordAdded, Toast.LENGTH_SHORT).show();
+            }
+            return CODE_UPDATED;
+        }
+
+        // Delete duplicates
+        UserDictionarySettings.deleteWordInInsertMode(newWord, resolver);
+
+        // In this class we use the empty string to represent 'all locales' and mLocale cannot
+        // be null. However the addWord method takes null to mean 'all locales'.
+        UserDictionary.Words.addWord(context, newWord,
+                Integer.parseInt(newWeight), newShortcut, TextUtils.isEmpty(mLocale) ?
+                        null : LocaleUtils.constructLocaleFromString(mLocale));
+
+        return CODE_WORD_ADDED;
+    }
+
+    public boolean isExistingWord(final Context context) {
+        final String newWord = mWordEditText.getText().toString();
+        if (mMode == MODE_INSERT || apply(context) == CODE_ALREADY_PRESENT) {
+            return hasWord(newWord, context);
+        } else {
+            return false;
+        }
+    }
+
+    private static final String[] HAS_WORD_PROJECTION = { UserDictionary.Words.WORD };
+    private static final String HAS_WORD_SELECTION = UserDictionary.Words.WORD + "=?";
+
+    private boolean hasWord(final String word, final Context context) {
+        final Cursor cursor;
+
+        cursor = context.getContentResolver().query(UserDictionary.Words.CONTENT_URI,
+                HAS_WORD_PROJECTION, HAS_WORD_SELECTION,
+                    new String[] { word }, null);
+
+        try {
+            if (null == cursor) return false;
+            return cursor.getCount() > 0;
+        } finally {
+            if (null != cursor) cursor.close();
+        }
+    }
+
+    public static class LocaleRenderer {
+        private final String mLocaleString;
+        private final String mDescription;
+
+        public LocaleRenderer(final Context context, @Nullable final String localeString) {
+            mLocaleString = localeString;
+
+            if (null == localeString || "".equals(localeString)) {
+                mDescription = context.getString(R.string.user_dict_settings_all_languages);
+            // Special treatment for the known languages with _zz and _ZZ types
+            } else if (localeString.endsWith("_zz") || localeString.endsWith("_ZZ")) {
+                final int resId = context.getResources().getIdentifier("subtype_"+localeString, "string", context.getPackageName());
+                mDescription = context.getString(resId);
+            } else {
+                mDescription = LocaleUtils.constructLocaleFromString(localeString).getDisplayName();
```

Comment:
Can't you just use `LocaleUtils.getLocaleDisplayNameInSystemLocale` here? Or does it give different results?

Reply:
Done with commit f7f791c
As we don't have access to _Locale_ I updated `UserDictionarySettings.getLocaleDisplayName()` instead and applied it where necessary.